pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,990 @@
+
+**Run this code** to print out `col1` from the DataFrame `df`.
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(df['col1'])
+</script>
+</lia-keep>
+</div>
+
+Did you notice the `Name` and `dtype` at the bottom? Those tell you the name of the series and its data type. Try changing `col1` to `col2` or `col3` to see what type of data those series contain.
+
+
+### Quiz: `pandas` package
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+d = {'col1': [1, 5, 7], 'col2': [3, .4, -2], 'col3':["yes", "no","blue"]};
+df = pd.DataFrame(data=d);
+print(df)
+</script>
+</lia-keep>
+</div>
+<br>
+Your friend's code (above) isn't running, which is extremely frustrating because they copied and pasted it from the previous page of this module. What line must they add to make it work?
+
+[[import pandas as pd]]
+[[?]] Hint: Run the code. What does the error message say is wrong?
+[[?]] Hint: What do we need to include at the top in order to make `pandas` commands available to us?
+***
+<div class = "answer">
+`NameError: name 'pd' is not defined` is a helpful error message because it tells you exactly what you friend forgot: they never defined `pd`!
+
+When you add the line `import pandas as pd` to the top of their code, it runs as it did before.
+
+<div class="python_run">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+d = {'col1': [1, 5, 7], 'col2': [3, .4, -2], 'col3':["yes", "no","blue"]};
+df = pd.DataFrame(data=d);
+print(df)
+</script>
+</lia-keep>
+</div>
+
+</div>
+***
+
+## Tabular Data in `pandas` DataFrames
+
+The tiny DataFrame we saw in the last section was exceptional in several ways. With real data you are likely to have some challenges that just can't be replicated with something that small:
+
+* Real data will almost always be too big to type into a DataFrame.
+
+* Real data frequently has too many rows and columns to look at the whole dataset at once.
+
+* Real data is often missing entries.
+
+### Loading data
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+</script>
+</lia-keep>
+</div>
+
+The `pandas` package can read most tabular data files and convert them into DataFrames. As long as your data is in place that the program can locate, either on your computer if you are running code on your computer, or in the cloud, all `pandas` needs to know is what type of file it is reading, and where to find that file.
+
+For this modules we will be looking at some fake Covid-19 testing data. This data is saved as a csv file and is hosted on [GitHub](https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv).
```

Comment:
This is optional, but my tendency with example data is to be abundantly clear which data are fake and which are real, so learners don't worry about potentially seeing PHI they're not supposed to (or maybe that's something only I worry about when I read data tutorials?). So I would add something like "...looking at some fake Covid-19 testing data. These are example data that were made up for teaching purposes so there's no real patient information here, but they are designed to mimic realistic testing data so the values and patterns are plausible."

Reply:
Good thought, I'm adding a box on the previous page laying this out explicitly.